homeworks/Homework5/TreeDAI/TreeDAI.cpp

```diff
@@ -0,0 +1,340 @@
+#include <iostream>
+
+using namespace std;
+
+enum Offenses
+{
+	speeding,
+	illegal_parking,
+	running_a_red_light,
+	no_seatbelt,
+	phone_use_while_driving,
+};
+
+class Car {
+	int carNumber;
+	Offenses* offense;
+	int size;
+
+public:
+	Car() {
+		this->carNumber = 1000;
+		this->size = 0;
+		this->offense = new Offenses[size];
+	}
+
+	Car(int number, int size, Offenses* offense) {
+		this->carNumber = number;
+		this->size = size;
+		this->offense = offense;
+	}
+
+	Car(int number) {
+		this->carNumber = number;
+		this->size = 0;
+		this->offense = new Offenses[size];
+	}
+
+
+	int getNumber() {
+		return this->carNumber;
+	}
+
+	int getSize() {
+		return this->size;
+	}
+
+	void addOffense(Offenses newOffense) {
+		Offenses* temp = new Offenses[size + 1];
+		for (size_t i = 0; i < size; i++)
+		{
+			temp[i] = offense[i];
+		}
+		temp[size] = newOffense;
+		delete[] offense;
+		offense = temp;
+		size++;
+	}
+
+	friend ostream& operator<<(ostream& out, Car car) {
+		out << "\nCar number: " << car.carNumber << endl;
+		out << "Offenses: \n";
+		for (size_t i = 0; i < car.size; i++)
+		{
+			switch (car.offense[i])
```

Comment:
I think it also can be siplifized, but not shure, tranform enum without switch
Try to google

Reply:
didn't found(